src/tokenizer.rs

```diff
@@ -562,7 +566,16 @@ impl<'a> Tokenizer<'a> {
                         s += s2.as_str();
                         return Ok(Some(Token::Number(s, false)));
                     }
-                    Ok(Some(Token::make_word(&s, None)))
+
+                    // https://dev.mysql.com/doc/refman/8.0/en/charset-introducer.html
+                    if ch == '_' && dialect_of!(self is MySqlDialect) {
```

Comment:
In general, I think it would be nice to support this also in the GenericDialect for consistency with other vendor specific features, but I don't think it is required.